src/v3/core/graph.test.js

```diff
@@ -847,6 +847,96 @@ describe("core/graph", () => {
     });
   });
 
+  describe("toJSON / fromJSON", () => {
+    const src = NodeAddress.fromParts(["src"]);
+    const dst = NodeAddress.fromParts(["dst"]);
+    const edge1 = () => ({
+      src,
+      dst,
+      address: EdgeAddress.fromParts(["edge"]),
+    });
+    const edge2 = () => ({
+      src: dst,
+      dst: src,
+      address: EdgeAddress.fromParts(["edge", "2"]),
+    });
+    it("toJSON matches snapshot", () => {
+      const graph = new Graph()
+        .addNode(src)
+        .addNode(dst)
+        .addEdge(edge1())
+        .addEdge(edge2());
+      expect(graph.toJSON()).toMatchSnapshot();
```

Comment:
Hmm. I appreciate that we have a snapshot test, but it would be nice if
that snapshot test could demonstrate some edge cases: multiple edges
with the same hom, loops, multiple loops, isolated nodes? It looks like
these cases aren’t tested at all, in fact.